main.cpp

```diff
@@ -389,16 +414,18 @@ void Turntable::changeSpeedSelection(bool isIncreased)
         speedSelection = 33.0f;
     }
     std::cout << "New Speed selection: " << speedSelection << "rpm\n";
+    regulateSpeed();
 }
 
-void Turntable::rotatePlatter()
+void Turntable::regulateSpeed()
 {
-    float speedDifference = platterRpm - speedSelection;
-    if (std::abs(speedDifference) > 0.05f)
+    while (true)
     {
-        platterRpm -= 0.1f*(speedDifference);
+        float speedDifference = speedSelection - platterRpm;
+        if (std::abs(speedDifference) < 0.05f) return;
```

Comment:
put this return statement on its own line so the logic is easy to see. 
It's easy to think that `platterRpm += 0.5` is the only thing that happens if that `if()` expression is true and just has bad indentation.

Again, the goal is to write clear, easy-to-understand code
- [ ] revise